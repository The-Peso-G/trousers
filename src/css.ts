import { StyleCollector } from './';
import { Expression, StyleDefinition } from './types';

export interface SingleStyleCollector<Theme> {
    get: () => StyleDefinition<{}, {}, Theme>[];
    getElementName: () => string;
}

export default function css<Theme = {}>(
    styles: TemplateStringsArray,
    ...expressions: Expression<Theme>[]
): SingleStyleCollector<Theme> {
    const styleCollector = new StyleCollector<{}, {}, Theme>('css');

    styleCollector.element(styles, ...expressions);

    return {
        get: () => styleCollector.get(),
        getElementName: () => styleCollector.getElementName(),
    };
}
