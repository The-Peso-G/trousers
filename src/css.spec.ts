import css from './css';

describe('css', () => {
    it('returns the correct element name', () => {
        const collector = css`
            font-style: normal;
        `;

        expect(collector.getElementName()).toEqual('css');
    });

    it('registers styles', () => {
        const collector = css`
            font-style: normal;
        `;

        expect(collector.get().length).toBe(1);
        expect(collector.get()[0].hash).toEqual('4061766009');
    });
});
