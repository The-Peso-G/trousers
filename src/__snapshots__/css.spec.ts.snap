// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css registers styles 1`] = `1502522116`;
